src/main/java/com/minecolonies/coremod/generation/defaults/DefaultResearchProvider.java

```diff
@@ -163,6 +168,22 @@ public DefaultResearchProvider(final DataGenerator generator)
         return researches;
     }
 
+    @Override
+    public JsonElement getBaseLanguageJson()
+    {
+        JsonElement langJson;
+        try
+        {
+            langJson =
+              new JsonParser().parse(new FileReader(this.generator.getInputFolders().stream().findFirst().orElse(null).resolve("assets/minecolonies/lang/en_us.json").toFile()));
```

Comment:
Prefer using `event.getExistingFileHelper()` to get a helper class to read existing files by ResourceLocation rather than by direct path.

Reply:
Is there a trick to doing so? It's giving errors about `@visiblefortesting` code in a production environment, ExistingFileHelper as a class is marked as Deprecated to be removed in 1.17, and when called in runData it doesn't seem like any of the minecolonies namespace is loaded into the resource manager.

Reply:
The class isn't deprecated, only one of the constructors is.  And you shouldn't be calling any of the constructors, you fetch the one that already exists inside the `GatherDataEvent`.  (Hence why I used an `event.` prefix.)

Reply:
Ok.  And `gatherDataEvent.getExistingFileHelper().getResource()` works for things in the minecraft, realms, or forge namespace, but that's all that's loaded in this context: is there something that needs be done to have the mod namespace loaded in an instance of `GatherDataEvent`?

Reply:
In the data run command in build.gradle, swap `--input` to `--existing`.  Optionally, also add this at the end to make the Structurize resources visible too:
```
, '--existing-mod', 'structurize'
```